### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by appleaze**
1. Added Category Image
2. Run Seeder on localdb
3. Run Homepage

SQL queries (if needed):
`INSERT INTO categories
VALUES (1, 'clothes', 'category_images/clothing.png'), (2, 'furniture', 'category_images/furniture.png');

INSERT INTO products
VALUES 
(1, 'Jacket', 'Sustainable jacket from wools', 250.000, 20, 1, 4.5),
(2, 'Shirt', 'Sustainable shirt from wools', 100.000, 15, 1, 4.8),
(3, 'Table', 'Sustainable table', 450.000, 10, 2, 4.8),
(4, 'Socks', 'Sustainable socks from wools', 50.000, 15, 1, 4.1),
(5, 'Chair', 'Sustainable chair', 300.000, 5, 2, 4.8);

INSERT INTO product_images VALUES
(1, 'product_images/wool_jacket.jpg', 1),
(2, 'product_images/wool_shirt.jpg', 2),
(3, 'product_images/table.jpg', 3),
(4, 'product_images/socks.jpg', 4),
(5, 'product_images/chair.jpg', 5);`

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class HomeController extends Controller
+{
+    /**
+     * Summary of getAllHome
+     * @param \App\Services\HomeService $homeService
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function getAllHome(HomeService $homeService)
```
Comment (IdjiotSandwiches) [inline app/Http/Controllers/HomeController.php]: ganti aj nama function ny jdi index

File: app/Services/HomeService.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Summary of getHomeProduct
+     * @return mixed
+     */
+    public function getHomeProduct()
+    {
+        //hitung product kelipatan 4
+        $totalProducts = Product::count();
+        $limit = $totalProducts - ($totalProducts%4);
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php]: Limit static aj gk sih lgsng limit 4, ato 8?

File: app/Services/HomeService.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            $limit = $totalProducts;
+        }
+        
+        $products = Product::take($limit)->get()
+        ->map(function($product) {
+            $name = $product->name;
+            $price = $product->price;
+            $rating = $product->sustainability_score;
+
+            $imgUrl = ProductImage::where('product_id', $product->id)->first();
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php]: Pake eager load aj, barnegan lgsng pas di model Product::with(...)

File: app/Services/HomeService.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            $link = route('getProduct', base64_encode("$name-$product->id"));
+
+            return (object) compact('name', 'rating', 'price', 'img', 'link');
+        });
+        return $products;
+    }
+    
+    /**
+     * Summary of getCategoryAll
+     * @return mixed
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php]: Ganti ke return object

File: app/Services/HomeService.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+     * Summary of __construct
+     */
+    public function __construct() 
+    {
+        $this->googleDriveUtility = new GoogleDriveUtility();
+    }
+
+    /**
+     * Summary of getHomeProduct
+     * @return mixed
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php]: Ganti return object

File: app/Services/HomeService.php [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        //hitung product kelipatan 4
+        $totalProducts = Product::count();
+        $limit = $totalProducts - ($totalProducts%4);
+        if($limit == 0 && $totalProducts >0){
+            $limit = $totalProducts;
+        }
+        
+        $products = Product::take($limit)->get()
+        ->map(function($product) {
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php:31]: image
  
Space ny benerin jdi gitu

File: app/Services/HomeService.php [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    
+    /**
+     * Summary of getCategoryAll
+     * @return mixed
+     */
+    public function getCategoryAll()
+    {
+        $categories = Category::take(4)->get()
+        ->map(function($category){
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php:53]: Benerin space ny

File: app/Services/HomeService.php [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        return $categories;
+    }
+
+    /**
+     * Summary of getBanner
+     * @return string[]
+     */
+    public function getBanner()
+    {
+        $bannerUrls = ['home_carousel_images/Banner1.png', 'home_carousel_images/Banner2.png', 'home_carousel_images/Banner3.png', 'home_carousel_images/Banner4.png'];
```
Comment (IdjiotSandwiches) [inline app/Services/HomeService.php:77]: Bisa pake cek storage yg cara GPT kemaren

File: resources/views/component/product-card.blade.php [hunk 9]
```diff
@@ -0,0 +1,17 @@
+@props(['link', 'image', 'name', 'price'])
```
Comment (IdjiotSandwiches) [inline resources/views/component/product-card.blade.php:1]: ini propsny bisa diganti jdi cuma product ny aj yg di pass

File: resources/views/component/tiles.blade.php
Comment (IdjiotSandwiches) [inline resources/views/component/tiles.blade.php:1]: Nama componentny diperjelas buat ap
category-tiles

File: resources/views/component/card.blade.php
Comment (IdjiotSandwiches) [inline resources/views/component/card.blade.php:1]: Ini nama fileny bisa diperjelas
product-card
klo emg ke double, yg lama apus aj, ato pake yg udh ad, tpi khusus yg file product-card sebelumny, itu hrs tetep propsny yg dipassing 1 1 soalny hrs dri JS, saran gw lu modif aj product-card yg udh ad

File: resources/views/vendor/pagination/bootstrap-4.blade.php
Comment (IdjiotSandwiches) [inline resources/views/vendor/pagination/bootstrap-4.blade.php:1]: btw file vendor buat pagination di hapus aj

File: tailwind.config.js
Comment (IdjiotSandwiches) [inline tailwind.config.js:1]: tw-elements ny hapus aj

**Review by IdjiotSandwiches — CHANGES_REQUESTED** [review]
Nanti klo udh lu perbaiki, gw coba refactor lgi buat nyesuain

File: app/Http/Controllers/HomeController.php (hunk 1 above)
Comment (appleaze) [reply app/Http/Controllers/HomeController.php]: resolved

File: app/Services/HomeService.php (hunk 2 above)
Comment (appleaze) [reply app/Services/HomeService.php]: Update: limit statci to 8 products

File: app/Services/HomeService.php (hunk 3 above)
Comment (appleaze) [reply app/Services/HomeService.php]: eager load implemented

File: app/Services/HomeService.php (hunk 4 above)
Comment (appleaze) [reply app/Services/HomeService.php]: resolved

File: app/Services/HomeService.php (hunk 5 above)
Comment (appleaze) [reply app/Services/HomeService.php]: resolved

File: app/Services/HomeService.php (hunk 6 above)
Comment (appleaze) [reply app/Services/HomeService.php:31]: resolved

File: app/Services/HomeService.php (hunk 7 above)
Comment (appleaze) [reply app/Services/HomeService.php:53]: resolved

File: resources/views/component/tiles.blade.php
Comment (appleaze) [reply resources/views/component/tiles.blade.php:1]: renamed to `category-tiles.blade.php`

File: resources/views/component/card.blade.php
Comment (appleaze) [reply resources/views/component/card.blade.php:1]: pake yang current props udah disamain

File: resources/views/welcome.blade.php [hunk 10]
```diff
@@ -1,54 +1,46 @@
...
+            <div class="grid grid-cols-1 sm:grid-cols-4 md:grid-cols-4 lg:grid-cols-4 gap-3">
+                @foreach ($categories as $category)
+                    @include('component.category-tiles', [
+                        'image' => $category->image,
+                        'name' => $category->name,
+                    ])
+                @endforeach
+            </div>
+        </div>
+
```
Comment (IdjiotSandwiches) [inline resources/views/welcome.blade.php]: benerin whitespace ny

**Review by IdjiotSandwiches — CHANGES_REQUESTED** [review]

File: resources/views/component/product-card.blade.php
Comment (IdjiotSandwiches) [inline resources/views/component/product-card.blade.php:1]: Knp tag a ny banyak bet?

File: resources/views/component/category-tiles.blade.php [hunk 11]
```diff
@@ -1,8 +1,10 @@
+@props(['image', 'name'])
```
Comment (IdjiotSandwiches) [inline resources/views/component/category-tiles.blade.php]: yg di passing bisa $category ny aj
jdi pass di componentny tinggal:
$category->name
$category->image

**Review by IdjiotSandwiches — CHANGES_REQUESTED** [review]

**Review by IdjiotSandwiches — APPROVED** [review]

**Pull request merged**